src/iop/colorbalancergb.c

```diff
@@ -423,7 +537,32 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
 
     Ych_to_gradingRGB(Ych, RGB, white_grading_RGB);
     dot_product(RGB, output_matrix, pix_out);
-    for(size_t c = 0; c < 4; ++c) pix_out[c] = fmaxf(pix_out[c], 0.f);
+
+    if(mask_display)
+    {
+      // draw checkerboard
+      float color;
+      if(i % checker_1 < i % checker_2)
+      {
+        if(j % checker_1 < j % checker_2) color = 1.0f;
+        else color = 0.5f;
+      }
+      else
+      {
+        if(j % checker_1 < j % checker_2) color = 0.5f;
+        else color = 1.0f;
+      }
+
+      float opacity = opacities[g->mask_type];
+      const float opacity_comp = 1.0f - opacity;
+
+      for(size_t c = 0; c < 4; ++c) pix_out[c] = opacity_comp * color + opacity * fmaxf(pix_out[c], 0.f);
```

Comment:
this for and the one below could be replaced by **for_four_channel()** and will ensure that SIMD inst are used if available.

Reply:
Right, I forgot about this one.

Reply:
Done.